Src/Plugins/ValidatorBase.cs

```diff
@@ -157,6 +157,33 @@ public static string ReturnUnhandledException(ref string message, Exception e, s
             return nameof(ValidationState.Unknown);
         }
 
+        /// <summary>
+        /// Normalized specific Shannon entropy. See https://rosettacode.org/wiki/Entropy
+        /// </summary>
+        public static double ShannonEntropy(string input, int countOfPossibleSymbols)
+        {
+            double entropy = 0;
+
+            if (string.IsNullOrWhiteSpace(input)) { return entropy; }
+
+            var charCounts = new Dictionary<char, double>();
+
+            foreach (char ch in input)
+            {
+                charCounts.TryGetValue(ch, out double count);
+                charCounts[ch] = ++count;
+            }
+
+            foreach (char ch in charCounts.Keys)
+            {
+                double count = charCounts[ch];
+                double frequency = count / (double)input.Length;
```

Comment:
>double [](start = 44, length = 6)

i think we dont need this cast

Reply:
Yes, that's true. Fixed.

---
In reply to: [577963751](https://github.com/microsoft/sarif-pattern-matcher/pull/225#discussion_r577963751) [](ancestors = 577963751)